lib/src/utils.dart

```diff
@@ -273,6 +273,196 @@ String getLineEnding(String yaml) {
   return windowsNewlines > unixNewlines ? '\r\n' : '\n';
 }
 
+/// Extracts comments for a node that is replaced within a [YamlMap] or
+/// [YamlList] or a top-level [YamlScalar] of the [yaml] string provided.
+///
+/// [currentEndOffset] represents the end offset of [YamlScalar] or [YamlList]
+/// or [YamlMap] being replaced, that is, `end + 1`.
+///
+/// [nextStartOffset] represents the start offset of the next [YamlNode].
+/// May be null if the current [YamlNode] being replaced is the last node
+/// in a [YamlScalar] or [YamlList] or if its the only top-level [YamlScalar].
```

Comment:
Just curious, but does this make assumptions about block-mode?

Reply:
I wrote that function to be open ended and versatile in case someone wanted to tackle #71 or do something cool with `YamlEditor` if they have the next node in a `YamlList` or `YamlMap`

Reply:
That was a heads-up for anyone who may decide to call that function. It should be `YamlMap` or `YamlList`.